homeassistant/components/here_weather/sensor.py

```diff
@@ -0,0 +1,116 @@
+                HEREDestinationWeatherSensor(
+                    entry,
+                    here_weather_data_dict[sensor_type].coordinator,
+                    sensor_type,
+                    weather_attribute,
+                )
+            )
+    async_add_entities(sensors_to_add)
+
+
+class HEREDestinationWeatherSensor(CoordinatorEntity, SensorEntity):
+    """Implementation of an HERE Destination Weather sensor."""
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator,
+        sensor_type: str,
+        weather_attribute: str,
+        sensor_number: int = 0,  # Additional supported offsets will be added in a separate PR
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._base_name = entry.data[CONF_NAME]
+        self._name_suffix = SENSOR_TYPES[sensor_type][weather_attribute]["name"]
+        self._latitude = entry.data[CONF_LATITUDE]
+        self._longitude = entry.data[CONF_LONGITUDE]
+        self._sensor_type = sensor_type
+        self._sensor_number = sensor_number
+        self._weather_attribute = weather_attribute
+        self._unit_of_measurement = convert_unit_of_measurement_if_needed(
+            self.coordinator.hass.config.units.name,
+            SENSOR_TYPES[sensor_type][weather_attribute]["unit_of_measurement"],
+        )
+        self._device_class = SENSOR_TYPES[sensor_type][weather_attribute][
+            "device_class"
+        ]
+        self._unique_id = "".join(
+            f"{self._latitude}_{self._longitude}_{self._sensor_type}_{self._name_suffix}_{self._sensor_number}".lower().split()
+        )
+        self._unique_device_id = "".join(
+            f"{self._latitude}_{self._longitude}_{self._sensor_type}".lower().split()
+        )
+
+    @property
+    def entity_registry_enabled_default(self):
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return False
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self._base_name} {self._sensor_type} {self._name_suffix} {self._sensor_number}"
+
+    @property
+    def unique_id(self) -> str:
+        """Set unique_id for sensor."""
+        return self._unique_id
+
+    @property
+    def state(self) -> StateType:
+        """Return the state of the device."""
+        return get_attribute_from_here_data(
+            self.coordinator.data,
+            self._weather_attribute,
+            self._sensor_number,
+        )
+
+    @property
+    def unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+
+        return {
+            "identifiers": {(DOMAIN, self._unique_device_id)},
```

Comment:
For weather services we normally group the entities by location or api key per service.

Reply:
So like `"identifiers": {(DOMAIN, f"{self._latitude}_{self._longitude}, self._api_key)},` ?

Reply:
Yes, that's ok, but use the same separator, underscore, in the whole string. Ie replace the comma.

Reply:
Now I get it. `_unique_device_id` already contains the location and the weather mode.
This way each "device" has one weather entity and the sensor entities.
So this should be okay?
If i omit the weather mode it would be 5 weather entities and all the sensor entities per "device"

Reply:
We don't have clear rules regarding this yet so both would be ok, I think.

Reply:
In this case I prefer the current implementation. Gives me a better overview/grouping